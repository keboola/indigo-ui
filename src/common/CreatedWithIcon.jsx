import React from 'react';
import date from '../../static/helpers/date.js';

export default React.createClass({
  displayName: 'CreatedWithIcon',

  propTypes: {
    createdTime: React.PropTypes.string
  },

  render: function() {
    return (
      <span>
          <i className="fa fa-fw fa-calendar" />
          {date.format(this.props.createdTime)}
      </span>
    );
  }
});
