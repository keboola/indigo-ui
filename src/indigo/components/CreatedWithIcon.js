import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import date from './date';

class CreatedWithIcon extends React.Component {
  render() {
    if (this.props.relative) {
      return (
        <span title={date.format(this.props.createdTime)}>
          <i className="fa fa-fw fa-calendar" /> {moment(this.props.createdTime).fromNow()}
        </span>
      );
    }
    return (
      <span title={moment(this.props.createdTime).fromNow()}>
        <i className="fa fa-fw fa-calendar" /> {date.format(this.props.createdTime)}
      </span>
    );
  }
}

CreatedWithIcon.propTypes = {
  createdTime: PropTypes.string.isRequired,
  relative: PropTypes.bool,
};

CreatedWithIcon.defaultProps = {
  relative: true,
};

export default CreatedWithIcon;
