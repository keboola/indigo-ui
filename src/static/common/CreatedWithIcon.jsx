import React from 'react';
import date from '../utils/date.js';

export default React.createClass({
  displayName: 'CreatedWithIcon',
  propTypes: {
    createdTime: React.PropTypes.string
  },
  render: function() {
    return (
      <span className="time-label time-label-created" title={date.fromNow(this.props.createdTime)}>
          <i className="fa fa-fw fa-calendar" />
          {date.format(this.props.createdTime)}
      </span>
    );
  }
});