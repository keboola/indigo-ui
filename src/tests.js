import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
