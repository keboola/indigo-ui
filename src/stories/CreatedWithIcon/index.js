import React from 'react';
import {storiesOf} from '@storybook/react';
import {withInfo} from '@storybook/addon-info';
import CreatedWithIcon from '../../indigo/components/CreatedWithIcon';

const demoDatetime = "2018-11-07 21:29:36";

storiesOf('CreatedWithIcon', module)
  .add(
    'Relative time',
    withInfo({
      inline: true,
    })(() => {
      return <CreatedWithIcon createdTime={demoDatetime} />;
    })
  )
  
  .add(
    'Absolute time',
    withInfo({
      inline: true,
    })(() => {
      return <CreatedWithIcon createdTime={demoDatetime} relative={false} />;
    })
  )
  
