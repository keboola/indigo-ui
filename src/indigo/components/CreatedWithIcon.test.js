import React from 'react';
import moment from 'moment';
import { snapshot } from '../../tests';
import CreatedWithIcon from './CreatedWithIcon';

const time1 = '2018-11-26T15:11:05';

const fakeTimeNow = moment(time1).valueOf();
global.Date.now = jest.fn(() => fakeTimeNow);
const timePast = moment(fakeTimeNow - 123456)
  .format('YYYY-MM-DDTHH:mm')
  .toString();

describe('<CreatedWithIcon />', () => {
  it('Relative time', () => {
    snapshot(<CreatedWithIcon createdTime={timePast} />);
  });

  it('Absolute time', () => {
    snapshot(<CreatedWithIcon createdTime={timePast} relative={false} />);
  });
});
