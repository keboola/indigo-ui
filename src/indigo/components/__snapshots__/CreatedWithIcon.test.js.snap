// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatedWithIcon /> Absolute time 1`] = `
<span
  title="2 minutes ago"
>
  <i
    className="fa fa-fw fa-calendar"
  />
   
  2018-11-26 15:09:00
</span>
`;

exports[`<CreatedWithIcon /> Relative time 1`] = `
<span
  title="2018-11-26 15:09:00"
>
  <i
    className="fa fa-fw fa-calendar"
  />
   
  2 minutes ago
</span>
`;
